app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,365 @@
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStates;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.Task;
+import java.lang.reflect.Method;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.wifi_hotspot.HotspotService;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.ZimFileSelectFragment;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.isStarted;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.stopAndroidWebServer;
+import static org.kiwix.kiwixmobile.wifi_hotspot.HotspotService.checkHotspotState;
+
+public class ZimHostActivity extends AppCompatActivity implements
+    ServerStateListener {
+
+  Button startServerButton;
+  TextView serverTextView;
+
+  public static final String ACTION_TURN_ON_AFTER_O = "Turn_on_hotspot_after_oreo";
+  public static final String ACTION_TURN_OFF_AFTER_O = "Turn_off_hotspot_after_oreo";
+  private final String IP_STATE_KEY = "ip_state_key";
+  private static final int MY_PERMISSIONS_ACCESS_FINE_LOCATION = 102;
+  private Intent serviceIntent;
+  private Task<LocationSettingsResponse> task;
+  boolean flag = false;
+  String ip;
+  String TAG = ZimHostActivity.this.getClass().getSimpleName();
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_zim_host);
+
+    setUpToolbar();
+
+    startServerButton = (Button) findViewById(R.id.startServerButton);
+    serverTextView = (TextView) findViewById(R.id.server_textView);
+
+    if (savedInstanceState != null) {
+      serverTextView.setText(
+          getString(R.string.server_started_message) + " " + savedInstanceState.getString(
+              IP_STATE_KEY));
+      startServerButton.setText(getString(R.string.stop_server_label));
+    }
+
+    FragmentManager fragmentManager = getSupportFragmentManager();
+    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+    ZimFileSelectFragment fragment = new ZimFileSelectFragment();
+    fragmentTransaction.add(R.id.frameLayoutServer, fragment);
+    fragmentTransaction.commit();
+
+    serviceIntent = new Intent(this, HotspotService.class);
+    Context context = this;
+
+    startServerButton.setOnClickListener(new View.OnClickListener() {
+      @Override public void onClick(View v) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          toggleHotspot();
+        } else {
+          //TO DO: show Dialog() + within that add check mobile Data check later.
+          //if (isMobileDataEnabled(context)) {
+          //  mobileDataDialog();
+          //} else {
+          if (flag) {
```

Comment:
flag? What does that mean?

Reply:
I just realised that it's a duplicate of `isStarted` that is a variable to show if server has been started. So, this has been removed.

Reply:
8e6edf9f65139d287fffda63a340618b504847dd